src/java.base/share/classes/java/util/ReverseOrderListView.java

```diff
@@ -296,18 +299,24 @@ public <T> T[] toArray(IntFunction<T[]> generator) {
 
     // copied from AbstractCollection
     public String toString() {
-        Iterator<E> it = iterator();
-        if (! it.hasNext())
-            return "[]";
-
-        StringBuilder sb = new StringBuilder();
-        sb.append('[');
-        for (;;) {
-            E e = it.next();
-            sb.append(e == this ? "(this Collection)" : e);
-            if (! it.hasNext())
-                return sb.append(']').toString();
-            sb.append(',').append(' ');
+        if (base instanceof ImmutableCollections.StableList<E> stableList) {
+            final StableValueImpl<E>[] reversed = ArraysSupport.reverse(
+                    Arrays.copyOf(stableList.delegates, stableList.delegates.length));
+            return StableUtil.renderElements(base, "Collection", reversed);
```

Comment:
StableList should provide its own SubList class that provide this implementation, instead of hacking the general-purpose view.